lib/serviceAccessConfig/nginx.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright (C) 2019 SUSE LLC
+#  All rights reserved.
+#
+#  This file is part of serviceAccessConfig
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""serviceAccessConfig plugin for nginx"""
+
+import ConfigParser
+import logging
+
+from accessrulegenerator import ServiceAccessGenerator
+from generatorexceptions import *
+
+
+class ServiceAccessGeneratorNginx(ServiceAccessGenerator):
+    """Specific access rule generator for nginx"""
+
+    # ======================================================================
+    def __init__(self, ip_source_config_file_name):
+        self.section_name = 'nginx'
+        super(ServiceAccessGeneratorNginx, self).__init__(
+            ip_source_config_file_name)
+
+    # ======================================================================
+    def _add_cidr_rules(self, content, cidr_blocks):
+        """Add the cidr bloc data to the content string"""
+
+        if content and not content.endswith('\n'):
+            content += '\n'
+
+        cidr_blocks = cidr_blocks.split(',')
+        header = '# ACL generated by serviceAccessConfig do not remove this\n'
+        header += '# comment entry from here to EOF will be over written\n'
```

Comment:
Rather that rewriting a section of the file, can we just rewrite the whole file? Apache config had other stuff in it, but for nginx/RMT it's not the main config file.

If something needs to be added manually it still would be possible to do the regular *nix-style config inclusion, e.g.:
* `00-manually-added-before.conf`
* `50-service-access.conf`
* `99-manually-added-after.conf`

Reply:
Well that works for us, but in general a third party user may want to combine access rules into 1 file and they might have some special rules outside the generator.

Reply:
Well, I'd still go for a simpler solution that works for us. It wouldn't in any way prevent 3rd party users from getting whatever they need done. (All the more reason to use includes if they have rules outside the generator)